.github/workflows/build-docker-image.yml

```diff
@@ -32,33 +35,65 @@ jobs:
       - name: go-test
         run: go test -v ./...
 
+  build-image-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: go version
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker info
+        run: docker info
+      - name: Buildx inspect
+        run: docker buildx inspect
+
+      - name: Build image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: linux/Dockerfile
+          platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v6
+          push: false
+          # Use a 'temp' tag, that won't be pushed for non-release builds
+          tags: ${{ env.IMAGE_REPOSITORY }}:${{ github.event.release.tag_name || 'temp' }}-linux
+
+  build-image-windows:
+    strategy:
+      matrix:
+        os-version:
+          - ltsc2019
+          - ltsc2022
+    runs-on: windows-2022
```

Comment:
Question: do we need to build on Win2022 for both OS versions, or it's fine to always use Win2022?

Reply:
I think @HofmeisterAn know better, how the host os is relevant for the resulting images.

That said, looking at the resulting multi-arch image, I can see that the image's os version is correct. I'm not entirely sure if the Golang compiler behaves differently dependent on the host os. In case we run into issues, I would try to be more explicit at the windows/Dockerfile (line 5 with `FROM golang:1.21-nanoserver as build`). For now I don't see a reason to change anything.

Reply:
In other words: the resulting images are using the desired os versions, but the compiled binary is the same. As long as Golang doesn't do strange things we're fine.

Reply:
> or it's fine to always use Win2022?

I think we do not need to distinguish, and it is fine to use the `windows-2022` runner for both versions. The `windows-2022` runner is capable of running both versions.

> In other words: the resulting images are using the desired/different os versions, but the compiled binary is the same. As long as Golang doesn't do strange things we're fine.

This is what I expected. LGTM.